tests/rptest/tests/cloud_retention_test.py

```diff
@@ -337,6 +356,7 @@ def cloud_log_size() -> int:
                 # for the duration of the test.
                 pass
             finally:
+                producer.wait()
```

Comment:
Should we `vassert(false)` after the wait on cloud size? Otherwise we may unintentionally skip the success path

Reply:
We should always hit the `finally`, but this was just confusing to read: I've added a commit that amends the test to avoid doing this weird "wait_until but expect a timeout" thing.